var/spack/repos/builtin/packages/qgis/package.py

```diff
@@ -128,6 +130,9 @@ class Qgis(CMakePackage):
     depends_on('qt@5.9.0:', when='@3.10.0:')
     depends_on('qtkeychain@:1.5', when='^qt@4')
     depends_on('qt@:4', when='@2')
+    # newer Python yields runtime error
+    # ref: https://github.com/spack/spack/issues/26803
+    depends_on('python@:3.8.6')
```

Comment:
Can you combine these with the `depends_on` above so that we don't specify Python version requirements in multiple places? It doesn't build with Python 3.8.7? Do we know why yet? Has this been reported upstream? We'll be switching the default Python version to 3.9 soon.

Reply:
for versions >=3.8.7, either QGIS hasn't built or else exhibited run time issues. I haven't reported this upstream since it is kind of normal that a given package doesn't build/work with the latest version of a dependency. I assume in the future QGIS will work with newer python versions. I will be getting rid of the extra `depends_on` though I would prefer a `conflicts` statement if nowadays it works as intended.

Reply:
> I haven't reported this upstream since it is kind of normal that a given package doesn't build/work with the latest version of a dependency.

I don't think this is expected, most packages work with the latest version of their dependencies. It's pretty rare to have to specify an upper bound on a dependency version in Spack, lower bounds are much more common.

> I would prefer a conflicts statement if nowadays it works as intended.

It works as intended with the new concretizer, which is now the default.